### PR TITLE
Move functions from ghdiff package to corehq.utils

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

Based on the [comment](https://github.com/dimagi/commcare-hq/pull/33044#issuecomment-1569917080), I realised that we can move the functions that we are using of `ghdiff` from the package to `corehq.utils` itself.

### Safety story

I have tested the changes locally and they were working fine. 
### Automated test coverage

Tests are added
### QA Plan
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by millerdev** [comment]
Under what license do we use ghdiff? Does it require attribution?

**Comment by AmitPhulera** [comment]
`ghdiff` uses MIT license and I have attributed the repository [here](https://github.com/dimagi/commcare-hq/pull/33126/commits/09b7a20f4729183042b62020edbc8a3d567918e0#diff-6ec99233d99b7885a1cf6a957c764983c48a7608398ca841b8ecfd9ac81c9939R1). Do you think it will work or we have to include author name and other info?

File: corehq/apps/app_manager/tests/test_ghdiff.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from django.test import SimpleTestCase
```
Comment (millerdev) [inline corehq/apps/app_manager/tests/test_ghdiff.py:1]: What do you think of moving this file into `corehq/util/tests`?

**Review by millerdev — COMMENTED** [review]
I wonder how we interpret this line?

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Would this be considered a "substantial portion of the software"? If yes, I think we would be required to copy the entire license into `corehq/util/ghdiff.py` and `test_ghdiff.py`.

**Comment by AmitPhulera** [comment]
> Would this be considered a "substantial portion of the software"? If yes, I think we would be required to copy the entire license into corehq/util/ghdiff.py and test_ghdiff.py.

I think so. I will add the license in the files.

**Review by millerdev — APPROVED** [review]

**Pull request merged**